site/source/docs/building_from_source/index.rst

```diff
@@ -11,23 +11,28 @@ Emscripten's core codebase, which is in the main "emscripten" repo, does not nee
 Building LLVM
 -------------
 
-For using the LLVM wasm backend (recommended), simply build normal upstream LLVM from the monorepo, including clang and wasm-ld (using something like ``-DLLVM_ENABLE_PROJECTS=lld;clang'``) and the wasm backend (which is included by default; just don't disable it), following `that project's instructions <http://llvm.org/docs/CMake.html>`_. For example, something like this can work:
+For using the LLVM wasm backend (recommended), simply build normal upstream LLVM from the `monorepo <https://github.com/llvm/llvm-project>`_, including clang and wasm-ld (using something like ``-DLLVM_ENABLE_PROJECTS='lld;clang'``) and the wasm backend (which is included by default; just don't disable it), following `that project's instructions <http://llvm.org/docs/CMake.html>`_. For example, something like this can work:
 
   ::
 
       mkdir build
-      cd build
-      cmake .. -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF
+      cd build/
+      cmake ../llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF
+      make -j$(nproc)  # no need to install
```

Comment:
I would replace this with `ninja` since it does automatic parallelism. 

And perhaps replace `no need to install` with a followup paragraph saying something like "Then setting your `LLVM_ROOT` in `.emscripten` to point to `<llvm_src>/build/bin`"

Reply:
Using ninja complexifies the cmake command, and doesn't seem to check for available memory, so I'll leave this part out (cmake --build .).

Reply:
I believe the ninja generator is smart enough to perform compilation with high level of parallelism and one a very low level (1 or 2 jobs in parallel) when does linking.    This is something that helps a lot with OOM and large links. . and something I don't think make can do at all.  Just one of many reasons to choose ninja.  Although you are right its probably out of scope for these instructions.